WebClock/Models/EfRepository/EfRepository.cs

```diff
@@ -24,6 +24,15 @@ public List<ClockInOut> Read(int id)
             var clockInOutDb = _context.ClocksInOut.Where(x => x.UserId == id);
             return clockInOutDb.Select(item => item.MapFromDb()).ToList();
         }
+        public ClockInOut ReadLast(int id)
+        {
+            var clocksInOut= _context.ClocksInOut.Where(x => x.UserId == id).OrderByDescending(x => x.Date).ToList();
+            if(clocksInOut.Count==0)
+            {
+                return null;
+            }
+            return clocksInOut.First().MapFromDb();
```

Comment:
Można to troszkę zrefaktoryzować:

            var lastClockInOut = _context.ClocksInOut.Where(x => x.UserId == id).OrderByDescending(x => x.Date).FirstOrDefault();
            return lastClockInOut?.MapFromDb();